docs/excel/excel-add-ins-ranges-advanced.md

```diff
@@ -64,6 +64,96 @@ Your add-in will have to format the ranges to display the dates in a more human-
 
 The `RangeAreas` object lets your add-in perform operations on multiple ranges at once. These ranges may be contiguous, but do not have to be. `RangeAreas` are further discussed in the article [Work with multiple ranges simultaneously in Excel add-ins](excel-add-ins-multiple-ranges.md).
 
+## Find special cells within a range (preview)
+
+The `Range.getSpecialCells()` and `Range.getSpecialCellsOrNullObject()` methods enable you to find at runtime the ranges that you want to operate on based on the characteristics of the cells and the type of the values of the cells. Both these methods return `RangeAreas` objects. Here are the signatures of the methods from the TypeScript data types file:
+
+```typescript
+getSpecialCells(cellType: Excel.SpecialCellType, cellValueType?: Excel.SpecialCellValueType): Excel.RangeAreas;
+```
+
+```typescript
+getSpecialCellsOrNullObject(cellType: Excel.SpecialCellType, cellValueType?: Excel.SpecialCellValueType): Excel.RangeAreas;
+```
+
+The following is an example of using the first one. About this code, note:
+
+- It limits the part of the sheet that needs to be searched by first calling `Worksheet.getUsedRange` and calling `getSpecialCells` for only that range.
+- It passes as a parameter to `getSpecialCells` the string version of a value from the `Excel.SpecialCellType` enum. Some of the other values that could be passed instead are "Blanks" for empty cells, "Constants" for cells with literal values instead of formulas, and "SameConditionalFormat" for cells that have the same conditional formatting as the first cell in the `usedRange`. The first cell is the upper leftmost cell. For a complete list of the values in the enum, see [beta office.d.ts](https://appsforoffice.microsoft.com/lib/beta/hosted/office.d.ts).
+- The `getSpecialCells` method returns a `RangeAreas` object, so all of the cells with formulas will be colored pink even if they are not all contiguous.
+
+```js
+Excel.run(function (context) {
+    var sheet = context.workbook.worksheets.getActiveWorksheet();
+    var usedRange = sheet.getUsedRange();
+    var formulaRanges = usedRange.getSpecialCells("Formulas");
```

Comment:
@kbrandl ExcelAPIs have unioned parameter types for the string literals corresponding to the names of the enums. I know @Zlatkovsky's preference is to use literals (hence why the type definitions even allow that). 

I don't think we have a standard of literal versus enum value in our docs. Should we?

Reply:
@AlexJerabek , @kbrandl :  Ah, this is another case where it's not just a question of standardizing the docs, but also having a stance on whether the person using this is using TypeScript or not.

For TypeScript (and hence also for Script Lab samples), I would absolutely use the strongly-typed strings, because you get all the benefits of compile-time safety and IntelliSense while having to type significantly less characters.

For JavaScript, I would stick with the actual Enum, because that's the only way to get IntelliSense and some semblance of ctrl+space safety.

As such, for articles (rather than Script Lab samples), I would probably go ahead and use the enums, since our audience is a mixed JS + TS crowd (and since TS is less prominent for us for now).

Reply:
@AlexJerabek -- I agree with @Zlatkovsky's suggestion that docs should use enums.